### ISSUE TITLE
List: deleting sub items requires deleting twice

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#40979]
### Description

When trying to delete sub items in the v2 of the List block, it requires deleting the block twice and you can only remove the block if you use the three dot menu (keyboard controls don't seem to work).  This was found as part of testing for the [FSE Outreach Program's fourteenth call for testing. ](https://make.wordpress.org/test/2022/04/28/fse-program-testing-call-14-rallying-recipe-reviewers/)

### Step-by-step reproduction instructions

1. Enable the List block experiment in Gutenberg > experiments.
2. Add a list block.
3. Add an initial item.
4. Add a sub item to that first item.
5. Try to delete it with your keyboard (you can't).
6. Try to delete it from the three dot ellipsis menu.
7. Notice that if there's content in the list item, it clears it but doesn't delete the list item.
8. Delete again and notice it removes the sub item finally.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/26996883/167718240-97fc0ea4-e664-4f08-98d1-e61db9e4e324.mov

https://user-images.githubusercontent.com/26996883/167718251-99495611-a019-4c76-a804-be4f84e8d7e8.mov

### Environment info

- WordPress 6.0 RC 2.
- GB Version 13.1.0 
- TT2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ellatrix** [comment]
#42503 also fixes this issue.

**Comment by annezazu** [comment]
Noting what this looks like in 6.3 RC1:

https://github.com/WordPress/gutenberg/assets/26996883/1c760dcb-3204-4a33-af97-90a0a5113752

**Comment by annezazu** [comment]
The PR hasn't yet been merged and with 6.5 Beta 1 next week, let's punt this.

**Comment by colorful-tones** [comment]
Hi folks,
We are only one week away from the Beta 1 cut-off date for [WordPress 6.6](https://make.wordpress.org/core/6-6/). This issue hasn’t seen any movement in a while, so we (the editor triage leads of the 6.6 release) have decided to remove it from the [WordPress 6.6 Editor Tasks project board](https://github.com/orgs/WordPress/projects/179/views/1).

**Comment by colorful-tones** [comment]
I don't think this is an issue anymore and we can likely close this out. See notes here: https://github.com/WordPress/gutenberg/pull/42503#issuecomment-2168313404
